agent/agent.go

```diff
@@ -106,17 +69,32 @@ func OpenDatabase(file string) (operator.Database, error) {
 }
 
 // NewLogAgent creates a new carbon log agent.
-func NewLogAgent(cfg *Config, logger *zap.SugaredLogger, operatorDir, databaseFile string) *LogAgent {
-	return &LogAgent{
-		Config:        cfg,
-		SugaredLogger: logger,
-		PluginDir:     operatorDir,
-		Database:      databaseFile,
-		buildParams:   make(map[string]interface{}),
+func NewLogAgent(cfg *Config, logger *zap.SugaredLogger, pluginDir, databaseFile string, buildParams map[string]interface{}) (*LogAgent, error) {
+	database, err := OpenDatabase(databaseFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "open database")
 	}
-}
 
-func (a *LogAgent) WithBuildParameter(key string, value interface{}) *LogAgent {
-	a.buildParams[key] = value
-	return a
+	registry, err := operator.NewPluginRegistry(pluginDir)
```

Comment:
@djaglowski I vaguely remember thinking we'd need to be able to override the plugins registry. Is it okay that we can't right now since we're making it and using it in the same function?

Reply:
Yeah, I don't recall a specific use case right now. Looks like we should be able to refactor if necessary though.